src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py

```diff
@@ -0,0 +1,39 @@
+"""Add user_id as a provided identity type
+
+Revision ID: 392992c7733a
+Revises: de456534dbda
+Create Date: 2023-01-20 15:33:24.084841
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "392992c7733a"
+down_revision = "de456534dbda"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.get_context().autocommit_block():
```

Comment:
Are these needed specifically? From my understanding, they automatically commit all the migrations before this operation, potentially prematurely.

> Warning As is necessary, the database transaction preceding the block is unconditionally committed. This means that the run of migrations preceding the operation will be committed, before the overall migration operation is complete.
It is recommended that when an application includes migrations with “autocommit” blocks, that [EnvironmentContext.transaction_per_migration](https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.params.transaction_per_migration) be used so that the calling environment is tuned to expect short per-file migrations whether or not one of them has an autocommit block.

Reply:
apparently the `ALTER TYPE` command [cannot be executed in a transaction](https://www.postgresql.org/docs/9.1/sql-altertype.html#AEN63173). I don't 100% understand it, but these answers go into some more detail: https://stackoverflow.com/questions/14845203/altering-an-enum-field-using-alembic

Reply:
That seems like it was for older postgres versions though

https://www.postgresql.org/docs/12/sql-altertype.html
> If ALTER TYPE ... ADD VALUE (the form that adds a new value to an enum type) is executed inside a transaction block, the new value cannot be used until after the transaction has been committed.

We do this in other migrations:
https://github.com/ethyca/fides/blob/main/src/fides/api/ctl/migrations/versions/906d7198df28_privacy_request_approve.py#L20

Reply:
oh great, thanks for finding that!!